ethereum/p2p/src/test/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/PeerDiscoveryControllerTest.java

```diff
@@ -647,18 +690,45 @@ public void shouldNotBondWithBlacklistedPeer()
         MockPacketDataFactory.mockNeighborsPacket(discoPeer, otherPeer, otherPeer2);
     controller.onMessage(neighborsPacket, discoPeer);
 
-    verify(controller, times(0)).bond(otherPeer, false);
-    verify(controller, times(1)).bond(otherPeer2, false);
+    verify(controller, times(0)).bond(otherPeer);
+    verify(controller, times(1)).bond(otherPeer2);
   }
 
   @Test
   public void shouldRespondToNeighborsRequestFromKnownPeer()
       throws InterruptedException, ExecutionException, TimeoutException {
-    final List<DiscoveryPeer> peers = createPeersInLastBucket(localPeer, 1);
+    final List<DiscoveryPeer> peers = new ArrayList<>();
+
+    // Flipping the most significant bit of the keccak256 will place the peer
+    // in the last bucket for the corresponding host peer.
+    final Bytes32 keccak256 = localPeer.keccak256();
+    final MutableBytesValue template = MutableBytesValue.create(keccak256.size());
+    byte msb = keccak256.get(0);
+    msb ^= MOST_SIGNFICANT_BIT_MASK;
+    template.set(0, msb);
+
+    /* * */
+
+    template.setInt(template.size() - 4, 0);
+    final Bytes32 keccak0 = Bytes32.leftPad(template.copy());
+    final MutableBytesValue id0 = MutableBytesValue.create(64);
+    UInt256.of(0).getBytes().copyTo(id0, id0.size() - UInt256Value.SIZE);
+    final DiscoveryPeer peer0 =
+        new DiscoveryPeer(
+            id0,
+            new Endpoint(
+                localPeer.getEndpoint().getHost(),
+                100 + counter.incrementAndGet(),
+                OptionalInt.empty()));
+    peer0.setKeccak256(keccak0);
+    peers.add(peer0);
+
+    /* * */
```

Comment:
Probably need to go back to having `createPeersInLastBucket` method here to hide the complex details of how to get a peer in the last bucket.